blenderproc/python/writer/GifWriterUtility.py

```diff
@@ -0,0 +1,210 @@
+                for index, perspective in enumerate(['_L', '_R']):
+                    tmp_dir = tempfile.TemporaryDirectory()
+                    print('Temporary directory for ' 
+                            + key 
+                            + perspective 
+                            + ': ' 
+                            + tmp_dir.name)
+                    to_animate.update({key + perspective: tmp_dir})
+                    for number, frame in enumerate(output_data_dict[key]):
+                        vis_data(key=key,
+                                data=frame[index], 
+                                save_to_file=tmp_dir.name + f'/{number}.png')
+            else:
+                # non stereo images
+                tmp_dir = tempfile.TemporaryDirectory()
+                print('Temporary directory for ' + key + ': ' + tmp_dir.name)
+                to_animate.update({key: tmp_dir})
+                for number, frame in enumerate(output_data_dict[key]):
+                    vis_data(key=key, 
+                            data=frame, 
+                            save_to_file=tmp_dir.name + f'/{number}.png')
+        return to_animate
+
+    @staticmethod
+    def _sort_method(item) -> int:
+        """ Order the images due to the file numbering """
+        _, file_name = item.rsplit('/', maxsplit=1)
+        file_number, _ = file_name.split('.')
+        return int(file_number)
+
+    @staticmethod
+    def _look_for_existing_output(output_dir_path: str, 
+                                append_to_existing_output: bool, 
+                                name_ending: str) -> int:
+        """ 
+        Looks for the highest existing #.gif number 
+        and adapts respectivly 
+        """
+        if append_to_existing_output:
+            gif_number = 0
+            for path in os.listdir(output_dir_path):
+                if path.endswith(name_ending):
+                    index = path[:-len(name_ending)]
+                    if index.isdigit():
+                        gif_number = max(gif_number, int(index) + 1)
+        else:
+            gif_number = 0
+        return gif_number
+
+    @staticmethod
+    def _write_to_gif(to_animate: Dict[str, str], 
+                    output_dir_path: str, 
+                    append_to_existing_output: bool, 
+                    time_constant: int, 
+                    time_direction: bool) -> None:
+        """
+        loads all .png files from each specific temporary folder 
+        and concatenates them to a single gif file respectively 
+        """
+        for key, tmp_dir in to_animate.items():
+            print('gif for ' + key + ' at ' + tmp_dir.name)
+            frames = []
+            # Loads all .png images as paths to a list
+            frame_paths = glob.glob(tmp_dir.name + '/*.png')
+            # sorts list in size w.r.t. given key
+            frame_paths.sort(key = GifWriterUtility._sort_method, 
+                            reverse=time_direction)
```

Comment:
Why are they loaded again, did we not just save them. Why did not we not save them in a list, then now we wouldn't have to sort them again.